runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/http/compression/CompressionAlgorithm.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package aws.smithy.kotlin.runtime.http.compression
+
+import aws.smithy.kotlin.runtime.io.SdkByteReadChannel
+import aws.smithy.kotlin.runtime.io.SdkSource
+
+/**
+ * Represents a compression algorithm. Used by an interceptor to compress request payloads on operations with the
+ * [requestCompression trait](https://smithy.io/2.0/spec/behavior-traits.html#requestcompression-trait)
+ */
+public interface CompressionAlgorithm {
+    /**
+     * The ID of the compression algorithm. This will be used to match against the supported compression algorithms
+     * of an operation.
+     */
+    public val id: String
+
+    /**
+     * The name of the content encoding to be appended to the `Content-Encoding` header.
+     * The [IANA](https://www.iana.org/assignments/http-parameters/http-parameters.xhtml)
+     * has a list of registered encodings for reference.
+     */
+    public val contentEncoding: String
+
+    /**
+     * Compresses a bytes array
+     * @return Compressed [ByteArray]
+     */
+    public fun compressBytes(bytes: ByteArray): ByteArray
```

Comment:
**fix**: If we're moving this to `http` then we can revert this back to `compress(request: HttpRequest): HttpRequest`.

If we want to leave this concept in `runtime-core` then I think we should refactor it to be `compress(stream: ByteStream): ByteStream` which will bring about some other changes to make that work for HTTP. We should talk about it.